dev-utils/gen-notice.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+#
+# Generate a NOTICE file for a distribution of this agent. It includes the
+# "./NOTICE.md" file (which includes the licenses of all vendored code), plus
+# the licenses for all included runtime dependencies.  This will also error out
+# if the license for a dep cannot be determined, or if the license isn't one of
+# the known ones.
+#
+# Usage:
+#   ./dev-utils/gen-notice.sh DIST_DIR
+#
+# where DIST_DIR is the distribution directory (the dir that holds the
+# "package.json" and "node_modules/" dir).
+#
+
+if [ "$TRACE" != "" ]; then
+    export PS4='${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+    set -o xtrace
+fi
+set -o errexit
+set -o pipefail
+
+# ---- support functions
+
+function fatal {
+    echo "$(basename $0): error: $*"
+    exit 1
+}
+
+# ---- mainline
+
+TOP=$(cd $(dirname $0)/../ >/dev/null; pwd)
+DIST_DIR="$1"
+[[ -n "$DIST_DIR" ]] || fatal "missing DIST_DIR argument"
+[[ -f "$DIST_DIR/package.json" ]] || fatal "invalid DIST_DIR: $DIST_DIR/package.json does not exist"
+
+# Directory holding some "license.*.txt" files for inclusion below.
+export LIC_DIR=$(cd $(dirname $0)/ >/dev/null; pwd)
+
+cat $TOP/NOTICE.md
+
+# Emit a Markdown section listing the license for each non-dev dependency
+# in the DIST_DIR. This errors out if a license cannot be found or isn't known.
+cd $DIST_DIR
+npm ls --omit=dev --all --parseable \
+    | node -e '
+        const fs = require("fs")
+        const path = require("path")
+        const knownLicTypes = {
+            "Apache-2.0": true,
+            "BSD-2-Clause": true,
+            "BSD-3-Clause": true,
+            "CC0-1.0": true,
+            "ISC": true,
+            "MIT": true,
+            "WTFPL OR ISC": true // oddball from is-integer package
+        }
+        const licFileNames = [
+            "LICENSE",
+            "LICENSE.txt",
+            "license.md",
+            "LICENSE-MIT",
+            "LICENSE-MIT.txt"
```

Comment:
The script worked great when I ran it on my case insensative Mac filesystem.  Will this   script ever run on linux build machines with case sensitive file systems?  

If so (I think?) we'll need the alternate casing of these file names (`license.md`, `License.md`, `LICENSE.md`) or we'll need something more complex that [the`existsSync` check here](https://github.com/elastic/apm-agent-nodejs/pull/2764/files#diff-57cd11bbaa52fcd7b495d503a57bd99e109b3a489322b031815d7883cfde6928R104) here.

Reply:
Nice catch. There were a few that didn't match that set when on a case-sensitive file system.

A little rabbit hole: npm uses [npm-packlist](https://www.npmjs.com/package/npm-packlist) to try to ensure a license file in a repo is included with `npm publish|pack`, which uses this logic: https://github.com/npm/npm-packlist/blob/da1ba4a3051ea822d9625347e00c186677247a45/lib/index.js#L20-L30

However that pattern does *not* include "LICENSE-MIT.*", which some of our current runtime deps use. I've changed the licFile handling to search the dir for a matching license file name that covers all these cases.

I've also added 'npm run lint:license-files' (added to `npm run lint`) that will run `gen-notice.sh ...` as part of linting. This means we'll find out if a new dep has a license issue for release before we get to release time.

commit 01de760b